### ISSUE TITLE
OCI Data Connector requires cursor group with static name but is not documented

### ISSUE THREAD

**Issue by PCNZ** [Azure/Azure-Sentinel#9203]
**Describe the bug**
Deploy the Oracle Cloud Infrastructure (OCI) data connector and function app as per the instructions. Function app cannot connect to OCI as the instruction's mis a step - the cursor group name in the OCI stream configuration must be hard coded to be group1.
https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Oracle%20Cloud%20Infrastructure/Data%20Connectors/AzureFunctionOCILogs/main.py
Within the python script is where the name group1 is hard coded.

**To Reproduce**
Deploy the OCI data connector as per instructions, get errors and function app does not collect logs.

**Expected behavior**
Steps in data connector and guides which mention what the cursor type value should be set to when deploying the function app and prior to this when creating the stream in OCI.
Ideally provide a way to specify the cursor group name as a variable for the function app - so it's not hard coded to group1.

Please update these pages with clear instructions.
https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Oracle%20Cloud%20Infrastructure/Data%20Connectors/azuredeploy_OCI_logs_API_FunctionApp.json
and
https://docs.oracle.com/en/learn/oci-logs-ms-azure-sentinel/index.html#introduction

**Screenshots**
Function app, 
New fields - cursor type and message limit - required but not documented.
![image](https://github.com/Azure/Azure-Sentinel/assets/68888744/4caa9276-366b-4f7e-9c0a-ccb7ad559c98)
![image](https://github.com/Azure/Azure-Sentinel/assets/68888744/ff42f069-d33d-486d-8b8e-2b55c6a5e2cc)

**Additional context**
Cursor type was added to the python code around January 2023 but the instructions were not updated.

**Comment by v-rbajaj** [comment]
Hi @PCNZ, thanks for flagging this, we will get back to you by 17 Oct 2023.

**Comment by v-muuppugund** [comment]
Hi @PCNZ ,After initial analysis, noticed that the issue exists and will work on issue resolution and ETA is 02Nov23

**Comment by v-muuppugund** [comment]
Hi @PCNZ , raised PR for it https://github.com/Azure/Azure-Sentinel/pull/9329 and waiting for approval, will update you once done.

**Comment by v-sudkharat** [comment]
Hi @PCNZ, Hope you are doing well. The PR review still in progress, we will share an update with you. Once the review is done.  Thanks!

**Comment by v-muuppugund** [comment]
Hi @PCNZ , This PR review still is in progress, we will share an update with you.once the review is done.Thanks!

**Comment by v-muuppugund** [comment]
@PCNZ The above PR is still under internal review,will post updates by 14Nov23

**Comment by v-muuppugund** [comment]
@PCNZ The above PR is still under internal review,will post updates by 17Nov23

**Comment by v-muuppugund** [comment]
@PCNZ ,The above PR is under internal review ,will post updates by 21Nov23

**Comment by v-muuppugund** [comment]
@PCNZ ,Working on internal review changes ,will get back to you 24Nov23

**Comment by v-muuppugund** [comment]
@PCNZ Will update you once the PR is completed

**Comment by v-muuppugund** [comment]
@PCNZ ,just want to update there are review comments for ARM template changes and code changes, so completed the changes, locally testing it, so once done. Will push to live, will keep you updated, Thanks.

**Comment by v-muuppugund** [comment]
@PCNZ  Apologies for the delayed response,will be sharing the package for today eod for testing ,so will proceed with the pushing it

**Comment by v-muuppugund** [comment]
> @PCNZ Apologies for the delayed response,will be sharing the package for today eod for testing ,so will proceed with the pushing it

@PCNZ yesterday didn't get chance to share the package,today will share it

**Comment by v-muuppugund** [comment]
> > @PCNZ Apologies for the delayed response,will be sharing the package for today eod for testing ,so will proceed with the pushing it
> 
> @PCNZ yesterday didn't get chance to share the package,today will share it


HI @PCNZ , Apologies for delay in my end as blocked with high priority work,Please  follow the below steps

1. Please use ARM template (https://github.com/Azure/Azure-Sentinel/blob/users/v-muppugundu/OCIDocumentationupdates/Solutions/Oracle%20Cloud%20Infrastructure/Data%20Connectors/azuredeploy_OCI_logs_API_FunctionApp.json)  and enter cursor type as group then Group Name and Group Instance Name mandatory
2. Deploy the new function app with Group instance and Group name created in oracle if needed we can have a call to explain in detail

**Comment by v-sudkharat** [comment]
Hi @PCNZ, I hope you are doing well. We are waiting for your response on above comment. Thanks!

**Comment by v-muuppugund** [comment]
Hi @PCNZ , Gentle Reminder,Could you please check on the above steps and share updates,I have scheduled call on Monday for Oracle AMA issue updates,We can discuss on that call,thanks.

**Comment by v-muuppugund** [comment]
Hi @PCNZ ,I have blocked calendar yesterday ,but unable to connect with you as you have declined ,Will block calendar again for testing this and close it from my end,Thanks.

**Comment by v-muuppugund** [comment]
Hi @PCNZ ,As discussed over teams, blocked calendar in teams, Please let me know if this time isn't convenient.

**Comment by v-muuppugund** [comment]
Hi @PCNZ ,as discussed yesterday over call, as you have don't have license and sdk based call,so need to test this package,will test with another customer and will update you once completed.

**Comment by v-muuppugund** [comment]
@PCNZ , today have a call with another customer who having oracle license ,will update you post the call.

**Comment by v-muuppugund** [comment]
@PCNZ , tested the issue with the cx i.e. (https://github.com/Azure/Azure-Sentinel/issues/10013) it worked, but unable to proceed to get results as cx having another issue,so will merge this pr with that issue i.e. ((https://github.com/Azure/Azure-Sentinel/issues/10013)),will update you once merged in the same issue,. If you still need support for 	this issue, feel free to re-open at any time. Thank you for your co-operation!
